Feat(signup): register a user

ah-med opened the pull request:
**What this PR does**
- create /auth/signup : register a user endpoint

**Tasks completed**
- write a test for endpoint
- add validation for user signup
- add  middleware for checking existing users
- create AuthController.signup

**Relevant pivotal tracker stories**
- [#160630063](https://www.pivotaltracker.com/story/show/160630063])

ah-med committed:
chore(database) - setup database

- add dotenv to dependencies
- add pg to dependencies
- add migrate command  to scripts

[Finishes #160630425]

ah-med committed:
chore(database) - setup database

- store database url in .env file
- store admin password in .env file

[Finishes #160630425]

ah-med committed:
chore(database) - setup database

- create a connection pool

[Finishes #160630425]

ah-med committed:
ore(database) - setup database

- create database schema
- seed the admin into the database

[Finishes #160630425]

ah-med committed:
chore(database) - setup database

- setup postgres for travis CI build

[Finishes #160630425]

ah-med committed:
chore(database) - setup database

- add database schema

[Finishes #160630425]

ah-med committed:
feat(signup) - add test for endpoint

- add test for signup endpoint

[Finishes #160630063]

ah-med committed:
feat(signup) - add middleware

- add a middleware to checkExisting accounts

[Finishes #160630063]

ah-med committed:
feat(signup) - add validations

- add validation schema for signup endpoint
- add validtion for signup

[Finishes #160630063]

ah-med committed:
feat(signup) - register a user

- add AuthContrller.sigup to contrllers
- add auth/signup route
- add serverErrors to errors
- update test script in package.json

[Finishes #160630063]

ChidinmaOrajiaku commented on server/controllers/AuthController.js:
```diff
@@ -0,0 +1,39 @@
+    const {
+      firstName, lastName, email, password
+    } = req.body;
+    const pass = bcrypt.hashSync(password, 10);
```
`const pass`? what is `pass`?

ChidinmaOrajiaku commented on server/controllers/AuthController.js:
```diff
@@ -0,0 +1,39 @@
+    const pass = bcrypt.hashSync(password, 10);
+
+    const text = 'INSERT INTO users(first_name, last_name, email, password, role) values($1, $2, $3, $4, $5) RETURNING user_id';
+    const param = [firstName, lastName, email, pass, 'user'];
```
`const param` is not a descriptive name o. `param` is too generic

ah-med committed:
feat(signup) - register a user

- change pass to hashedPassword
- change param to parameters

[Finishes #160630063]

ChidinmaOrajiaku commented:
lgtm
